daemon/emer-persistent-cache.c

```diff
@@ -459,7 +459,13 @@ update_boot_offset (EmerPersistentCache *self,
         return save_timing_metadata (self, &relative_time, &absolute_time, NULL,
                                      NULL, NULL, error);
 
-      return TRUE;
+      /*
+       * Always make sure a file exists with the boot offset to maintain the
+       * invariant that when this function returns TRUE, a keyfile exists on
+       * disk.
+       */
+      return save_timing_metadata (self, NULL, NULL, &priv->boot_offset,
```

Comment:
Just realized why this is happening. I wasn't careful enough in my previous review. It's the test that is wrong, not the production code. We can drop commit 9a06e479ee6d822ba52d35d4c07d9e1be28c252f and modify test_persistent_cache_get_offset_can_build_boot_metadata_file so that it calls `make_testing_cache` and unrefs the persistent cache, then unlinks the boot offset metadata file, and then calls `make_testing_cache` again. I am pretty comfortable only handling the case where the boot offset metadata file mysteriously went missing at initialization especially since we already write the boot offset metadata file hourly and at shutdown. Arguably, we don't need to handle it at all, but that would really explode the scope of this PR.